### ISSUE TITLE
bug: SQS receive messages with --max-number-of-messages unexpected behavior?

### ISSUE THREAD

**Issue by manuelwaltschek** [localstack/localstack#6695]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am testing localstack:1.0.4 SQS receive messages

When I execute sqs-receive-messages, 
I always receive one message in the "Messages" array. Therefore it seems I need to add  --max-number-of-messages to receive multiple messages at once. 

When I run it without max-number then I receive every message I send after each other and then it does not return any more.
But when I add the parameter it appears to be returning messages randomly, even when I execute purge-queue as seen below.

How should it behave? Why do I see these "ghost messages"?
There is also a random amount of received messages it seems.

### Expected Behavior

Receive only messages that have not been purged?
Receive messages only once even with max-number?
Receive a consistent amount of messages?

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

```
version: "3.8"

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack:1.0.4
    ports:
      - "127.0.0.1:4566:4566"            # LocalStack Gateway
      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
      # - "127.0.0.1:53:53"                # DNS config (only required for Pro)
      # - "127.0.0.1:53:53/udp"            # DNS config (only required for Pro)
      # - "127.0.0.1:443:443"              # LocalStack HTTPS Gateway (only required for Pro)
    environment:
      - SERVICES=${LOCALSTACK_SERVICES:-sqs,sns}
      - DEBUG=1 #${DEBUG-}
      # - PERSISTENCE=1 only working in pro version
      - DATA_DIR=${LOCALSTACK_DATA_DIR:-/tmp/localstack/data}
      - AWS_DEFAULT_REGION=eu-west-1
      #- LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR-}
      #- LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}  # only required for Pro
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"

```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Setup:

```
localstack-configure:
	aws configure set aws_access_key_id "dummy" --profile test-profile; \
  aws configure set aws_secret_access_key "dummy" --profile test-profile; \
  aws configure set region "eu-central-1" --profile test-profile; \
```

Creating queue with:

```
sqs-create-queue:
	aws --endpoint-url=http://localhost:4566 sqs create-queue --queue-name test-queue --profile test-profile --region eu-central-1 | cat
```

Sending messages with:

```
sqs-send-message:
	aws --endpoint-url=http://localhost:4566 sqs send-message  --queue-url http://localhost:4566/000000000000/test-queue --profile test-profile --region eu-central-1  --message-body '{ \
            "event_id": "7456c8ee-949d-4100-a0c6-6ae8e581ae15", \
            "event_time": "2019-11-26T16:00:47Z", \
            "data": { \
              "test": 83411 \
          } \
        }' | cat
```

```
sqs-receive-messages:
	aws --endpoint-url=http://localhost:4566 sqs receive-message --queue-url http://localhost:4566/000000000000/test-queue --max-number-of-messages 10 --profile test-profile --region eu-central-1 --output json | cat

```

```
sqs-purge-queue:
	aws --endpoint-url=http://localhost:4566 sqs purge-queue --queue-url http://localhost:4566/000000000000/test-queue --profile test-profile --region eu-central-1 --output json | cat

```

### Environment

```markdown
- OS: macOS Monterey 12.4
- LocalStack: docker image 1.0.4
```

**Comment by thrau** [comment]
Hi Manuel @manuelwaltschek, thanks for reporting. I think there are two things at play here.

The problem related to messages appearing in the queue after purge_queue is fixed in #6710. We weren't correctly clearing the in-flight messages, so messages that were received would re-appear after the visibility timeout expires.

Regarding `--max-number-of-messages` and the number of messages returned by receive-message: I don't think LocalStack is behaving wrongly here. Here's the API reference: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html

> When I execute sqs-receive-messages, I always receive one message in the "Messages" array. Therefore it seems I need to add --max-number-of-messages to receive multiple messages at once.

That is expected behavior, since the default value of `--max-number-of-messages` is 1.

> When I run it without max-number then I receive every message I send after each other and then it does not return any more.
But when I add the parameter it appears to be returning messages randomly, even when I execute purge-queue as seen below.

There are no guarantees about the ordering of message in standard queues, you need to use FIFO queues for that. The behavior with purge queue was indeed wrong as stated above.

> There is also a random amount of received messages it seems.
Yes, SQS provides no guarantees about how many messages you will receive, other than the upper bound of max-number-of-messages. See the docs: "If the number of messages in the queue is extremely small, you might not receive any messages in a particular ReceiveMessage response."

>  Receive messages only once even with max-number?
Maybe [this document on SQS message visibility timeout](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) can help clear up some of the confusion.

**Comment by thrau** [comment]
closing this since #6710 should fix the purge queue problem, and as i mentioned in my post from two weeks ago, the other behavior should be correct.

please open a new issue if the problem persists.
